### PR TITLE
Date range selection on Discover Events screen

### PR DESCRIPTION
**Pull request by QuicheLorraine3600**
This PR adds the ability to select a date range (not just a date) on the Discover Event screen.

### New UI component: DateRangeSelector
This component is very similate to DateSelector but offers the ability to select a range instead of a single date. 
You cannot select past dates (I didn't see where it could be useful in our project).

image
image

## Refactor of  `happensOnThisDateFilter` to `happensInDateRangeFilter`
`happensInDateRangeFilter` takes 2 date instead of one. `happensOnThisDateFilter` has been removed as it should not be used anymore in our project. If you *really* want to have something similar, just call  `happensInDateRangeFilter(day, day)`

## Modification of FindEventViewModel
I removed the `selectedDate` field of this view model to replace with `startDate` and `endDate`. The method `updateSelectedDate` has been replaced with `updateDateRange` accordingly.

### CODE REVIEW

**Review by leagrieder — CHANGES_REQUESTED** [review]
Great work on adding this feature, I think it makes alot of sense to be able to give users that functionality! 
The code changes are logical, well-documented, and thoroughly tested. The new `DateRangeSelector` component is a valuable addition that improves user experience on the Discover Event screen. 

There are just 2 strange things happening on the front end of the app when I tested it on an android phone and on the emulator. The first weird thing is the appearance of the text when we select a date range. The other strange thing is that some of the selected dates are not highlighted, was this wanted? 

image
image

File: app/src/main/res/values-fr/strings.xml [hunk 1]
```diff
@@ -51,6 +51,7 @@
     <string name="accountEditScreenButton">Modifier le compte</string>
     <string name="chimpagne_cancel">Annuler</string>
     <string name="chimpagne_add">Ajouter</string>
+    <string name="chimpagne_ok">Ok</string>
```
Comment (Athanaze) [inline app/src/main/res/values-fr/strings.xml:54]: OK

File: app/src/main/res/values-fr/strings.xml [hunk 2]
```diff
@@ -109,4 +110,7 @@
     <string name="account_created_toast">Compte créé</string>
     <string name="acount_creation_failed_toast">Création de compte échouée</string>
 
+    <!-- DateRangeSelector -->
+    <string name="select_date_range_for_query">Selectionnez une période pour la recherche</string>
```
Comment (Athanaze) [inline app/src/main/res/values-fr/strings.xml:114]: typo, should be Sélectionnez

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -75,6 +75,7 @@
     <string name="event_details_screen_car_pooling_button">Car pooling</string>
     <string name="chimpagne_cancel">Cancel</string>
     <string name="chimpagne_add">Add</string>
+    <string name="chimpagne_ok">Ok</string>
```
Comment (Athanaze) [inline app/src/main/res/values/strings.xml:78]: OK

File: app/src/androidTest/java/com/monkeyteam/chimpagne/newtests/model/event/QueryFiltersTests.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    var loading = true
+    var fetchedEvents = listOf<ChimpagneEvent>()
+    database.eventManager.getAllEventsByFilterAroundLocation(
+        events[0].location,
+        5000.0,
+        {
+          fetchedEvents = it
+          loading = false
+        },
+        { assertTrue(false) },
```
Comment (Athanaze) [inline app/src/androidTest/java/com/monkeyteam/chimpagne/newtests/model/event/QueryFiltersTests.kt:54]: There must be a better way

Comment (QuicheLorraine3600) [reply app/src/androidTest/java/com/monkeyteam/chimpagne/newtests/model/event/QueryFiltersTests.kt:54]: you seems to discover how tests are implemented in the project :monocle_face:

**Comment by QuicheLorraine3600** [comment]
Issues fixed:
image

**Review by juaniac — APPROVED** [review]
Very good PR.
Concise, self contained and brings the new functionality of a date range selector.
The code is structured, readable and well tested.
Thus I approve this PR.

You are however free to implement this suggested change to make the UI component a bit more intuitive to use:
Instead of selecting the start and end range dates one after the other (always in that order, for example in your design you cannot re-assign the end date twice in a row), it would be nice if it would automatically assign the later selected date as the end date for the range and the earlier one as the start date for the range.

**Pull request merged**